diskmaker/controllers/lv/reconcile.go

```diff
@@ -210,6 +260,30 @@ func (r *LocalVolumeReconciler) Reconcile(ctx context.Context, request ctrl.Requ
 	r.Log = r.Log.WithValues("request.namespace", request.Namespace, "Request.Name", request.Name)
 	r.Log.Info("Reconciling LocalVolume")
 
+	if !r.cacheSynced {
+		r.runtimeConfig.Node = &corev1.Node{}
+		err := r.Client.Get(ctx, types.NamespacedName{Name: os.Getenv("MY_NODE_NAME")}, r.runtimeConfig.Node)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		r.runtimeConfig.Name = common.GetProvisionedByValue(*r.runtimeConfig.Node)
+
+		pvList := &corev1.PersistentVolumeList{}
+		err = r.Client.List(context.TODO(), pvList)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to initialize PV cache: %w", err)
+		}
+		for _, pv := range pvList.Items {
+			// skip non-owned PVs
+			name, found := pv.Annotations[provCommon.AnnProvisionedBy]
+			if !found || name != r.runtimeConfig.Name {
+				continue
+			}
+			addOrUpdatePV(r.runtimeConfig, pv)
```

Comment:
When we're populating the cache here, is it possible to get a competing `addOrUpdatePV` call via `CreateFunc`?
https://github.com/openshift/local-storage-operator/blob/3defd5eb65f5e36e0009f5775225965eeecfeb11/diskmaker/controllers/lv/reconcile.go#L678-L683
If so, what would happen in that case? Could we end up with the same PV added twice?

Reply:
The PVs in cache are keyed by volume names, which is supposed to be a unique string cluster-wide and hence last update will win I think. For our purpose here, it won't matter I think.